### PR TITLE
test: Forge::addColumn() and null

### PR DESCRIPTION
**Pull request by kenjis**
~~Needs #7301~~

**Description**
See #7235

- add test for `Forge::addColumn()` and null

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: tests/system/Database/Live/ForgeTest.php [hunk 1]
```diff
@@ -825,6 +826,41 @@ public function testAddColumn()
...
+        $this->db->resetDataCache();
+
+        $col1 = $this->getMetaData('col1', 'forge_test_table');
+        $this->assertFalse($col1->nullable);
+        $col2 = $this->getMetaData('col2', 'forge_test_table');
+        $this->assertTrue($col2->nullable);
+        $col3 = $this->getMetaData('col3', 'forge_test_table');
+        $this->assertFalse($col3->nullable);
+        $col4 = $this->getMetaData('col4', 'forge_test_table');
+        $this->assertTrue($col4->nullable);
```
Comment (michalsn) [inline tests/system/Database/Live/ForgeTest.php:855]: Since `$col1` is `false` from a logical point of view, I think `$col4` should be `false` too. Thoughts?

Comment (kenjis) [reply tests/system/Database/Live/ForgeTest.php:855]: See #7235
I thought that, but if we change it, it is a breaking change.

Comment (michalsn) [reply tests/system/Database/Live/ForgeTest.php:855]: Yes, fair point but in this case, I would treat it as a bug. 
API should be predictable and having a different behavior for the "same thing" is something we should avoid.

Comment (michalsn) [reply tests/system/Database/Live/ForgeTest.php:855]: Okay, I'll be fine with documenting this and leaving it to be fixed in the next major version.

@codeigniter4/database-team any thoughts on this one?

Comment (kenjis) [reply tests/system/Database/Live/ForgeTest.php:855]: It is inconsistent to have different behavior when creating and adding, but I am a little doubtful that it is worth changing in v5.

However, if we decide to change it, I would like to include that in the documentation.

@codeigniter4/database-team Opinions?

Comment (MGatner) [reply tests/system/Database/Live/ForgeTest.php:855]: I am ambivalent.

**Comment by kenjis** [comment]
Rebased to resolve conflicts.

File: tests/system/Database/Live/ForgeTest.php (hunk 1 above)
Comment (kenjis) [reply tests/system/Database/Live/ForgeTest.php:856]: It seems nobody has strong opinion.

**Comment by kenjis** [comment]
This PR just adds test for the current implementation.
Merge.
